includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,312 @@
+	 */
+	protected function get_options( $assoc_args ) {
+		$defaults = array(
+			'checks'          => '',
+			'format'          => 'table',
+			'ignore-warnings' => false,
+			'ignore-errors'   => false,
+		);
+
+		$options = wp_parse_args( $assoc_args, $defaults );
+
+		if ( ! in_array( $options['format'], $this->output_formats, true ) ) {
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Output formats.
+					__( 'Invalid format argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->output_formats )
+				)
+			);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Gets the formatter instance to format check results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args Associative arguments.
+	 * @return WP_CLI\Formatter The formatter instance.
+	 */
+	protected function get_formatter( $assoc_args ) {
+		$default_fields = array(
+			'line',
+			'column',
+			'code',
+			'message',
+		);
+
+		if ( isset( $assoc_args['fields'] ) ) {
+			$default_fields = wp_parse_args( $assoc_args['fields'], $default_fields );
+		}
+
+		// If both errors and warnings are included, display the type of each result too.
+		if ( empty( $assoc_args['ignore_errors'] ) && empty( $assoc_args['ignore_warnings'] ) ) {
+			$default_fields = array(
+				'line',
+				'column',
+				'type',
+				'code',
+				'message',
+			);
+		}
+
+		return new WP_CLI\Formatter(
+			$assoc_args,
+			$default_fields
+		);
+	}
+
+	/**
+	 * Flattens and combines the given associative array of file errors and file warnings into a two-dimensional array.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $file_errors   Errors from a Check_Result, for a specific file.
+	 * @param array $file_warnings Warnings from a Check_Result, for a specific file.
+	 * @return array Combined file results.
+	 */
+	protected function flatten_file_results( $file_errors, $file_warnings ) {
+		$file_results = array();
+
+		foreach ( $file_errors as $line => $line_errors ) {
+			foreach ( $line_errors as $column => $column_errors ) {
+				foreach ( $column_errors as $column_error ) {
+
+					$file_results[] = array_merge(
+						$column_error,
+						array(
+							'type'   => 'ERROR',
+							'line'   => $line,
+							'column' => $column,
+						)
+					);
+				}
+			}
+		}
+
+		foreach ( $file_warnings as $line => $line_warnings ) {
+			foreach ( $line_warnings as $column => $column_warnings ) {
+				foreach ( $column_warnings as $column_warning ) {
+
+					$file_results[] = array_merge(
+						$column_warning,
+						array(
+							'type'   => 'WARNING',
+							'line'   => $line,
+							'column' => $column,
+						)
+					);
+				}
+			}
+		}
+
+		usort(
+			$file_results,
+			function( $a, $b ) {
+				if ( $a['line'] < $b['line'] ) {
+					return -1;
+				}
+				if ( $a['line'] > $b['line'] ) {
+					return 1;
+				}
+				if ( $a['column'] < $b['column'] ) {
+					return -1;
+				}
+				if ( $a['column'] > $b['column'] ) {
+					return 1;
+				}
```

Comment:
Can we combine the conditions? WDYT?

```suggestion
				if ( ( $a['line'] < $b['line'] ) || ( $a['column'] < $b['column'] ) ) {
					return -1;
				}
				if ( ( $a['line'] > $b['line'] ) || ( $a['column'] > $b['column'] ) ) {
					return 1;
				}
```

Reply:
Thanks @mukeshpanchal27 I don't think we can as we want to sort by line first and then column. So if we combine them I believe this could result in ordering items incorrectly.

Reply:
+1, also combining clauses is not always good for readability. Seeing the comparisons one by one after each other is probably easier to parse.